core/components/broker/broker.go

```diff
@@ -270,6 +279,12 @@ func (b component) HandleData(bctx context.Context, req *core.DataBrokerReq) (*c
 		}
 
 		if ok {
+			if counterReset {
+				if !entry.DevMode {
+					continue
```

Comment:
Doing this prevent the `mEntry` to be assigned + all the logging and analytic that happen below. 
This implies that in case of reset, no MIC check will ever succeed and the packet will just be ignored.